### PR TITLE
Add support for adding a manual (ssh) machine

### PR DESCRIPTION
**Pull request by AdamIsrael**
As discussed in issue #106, the model.add_machine method does not work when trying to add a manually instantiated machine with ssh credentials.

The reason for this is that the code that sets up and provisions the Juju agent on the target machine executes from the machine where the controller is bootstrapped, via the Juju CLI, and not the API. Part of this approach is because the ssh keys that the Juju controller uses to perform these operations only exist, by default, on the machine -- in `~/.local/share/juju/ssh` -- and not on the controller itself.

This PR re-implements a Python version of the process taken by the Juju code in `https://github.com/juju/juju/blob/develop/environs/manual/sshprovisioner/sshprovisioner.go`.

By necessity, I've changed the parameter required to add a manual machine, as it needs to know where the ssh keys are located. In cases where libjuju is being run from a machine different than the bootstrap host, the Juju keypair would need to be copied to it, or a new keypair made available and injected into the manually provisioned machine to enlist in Juju.

### CODE REVIEW

File: juju/provisioner.py [hunk 1]
```diff
@@ -0,0 +1,364 @@
+from .client import client
+
+import paramiko
```
Comment (tvansteenburgh) [inline juju/provisioner.py]: Need to add paramiko as a dep?

Comment (AdamIsrael) [reply juju/provisioner.py]: Added!

**Comment by tvansteenburgh** [comment]
It would be great to add a test for this, using a lxc as the machine to add.

File: juju/provisioner.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+        """
+
+        if isinstance(cmd, str):
+            cmd = shlex.split(cmd)
+
+        if type(cmd) is not list:
+            cmd = [cmd]
+
+        cmds = ' '.join(cmd)
+        stdin, stdout, stderr = ssh.exec_command(cmds, get_pty=pty)
```
Comment (johnsca) [inline juju/provisioner.py]: Not worth holding this PR over, but worth looking in to [asyncssh](https://github.com/ronf/asyncssh#asyncssh-asynchronous-ssh-for-python) to do the SSH work asynchronously.

**Pull request merged**